### PR TITLE
Update google api client to v25.x, enable code challenge/verifier

### PR DESCRIPTION
**Pull request by sqrrrl**
See https://developers.google.com/identity/protocols/OAuth2InstalledApp#step1-code-verifier for details about the protocol change.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -256,26 +256,23 @@ function getAPICredentials(cb) {
...
  function authorize(useLocalhost) {
+  let codes = oauth2Client.generateCodeVerifier();
   let opts = {
     access_type: 'offline',
     scope: [
       'https://www.googleapis.com/auth/script.deployments',
       'https://www.googleapis.com/auth/script.projects',
     ],
-  }
+    code_challenge_method: 'S256',
```
Comment (grant) [inline index.js]: Please include a link to:
https://developers.google.com/identity/protocols/OAuth2InstalledApp#step1-code-verifier

File: index.js [hunk 2]
```diff
@@ -256,26 +256,23 @@ function getAPICredentials(cb) {
...
-  authCode.then(code => {
-    return new Promise((res, rej) => {
-      oauth2Client.getToken(code, (err, token) => {
-        if (err) return rej(err);
-        return res(token);
-      })
-    })
-  })
-  .then(token => DOTFILE.RC.write(token))
+  authCode.then(code => oauth2Client.getToken({ code: code, codeVerifier: codes.codeVerifier }))
```
Comment (grant) [inline index.js]: - Destructure when possible.
- Line too long

```js
authCode.then(code => oauth2Client.getToken({
	code,
	codeVerifier: codes.codeVerifier
}))
```

File: index.js [hunk 3]
```diff
@@ -442,7 +439,7 @@ program
           if (error) {
             logError(error, ERROR.CREATE);
           } else {
-            let scriptId = res.scriptId;
+            let scriptId = res.data.scriptId;
```
Comment (grant) [inline index.js]: Please destructure `data` here and elsewhere in the function callback above.

```js
- script.projects.create({ title, parentId }, {}, (error, res) => {
+ script.projects.create({ title, parentId }, {}, (error, {data}) => {
```

File: index.js (hunk 1 above)
Comment (sqrrrl) [reply index.js]: done

File: index.js (hunk 2 above)
Comment (sqrrrl) [reply index.js]: done

File: index.js (hunk 3 above)
Comment (sqrrrl) [reply index.js]: done

**Comment by sqrrrl** [comment]
Updated

**Comment by grant** [comment]
Thanks! Will push to `npm` next week.

**Pull request merged**